### ISSUE TITLE
[TypeError: Cannot read property 'getRandomBase64String' of undefined] ERROR

### ISSUE THREAD

**Issue by alyiev** [aws/aws-sdk-js-v3#4680]
### Checkboxes for prior research

- [X] #4681
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Using "expo": "~48.0.15".

Getting this error:

[TypeError: Cannot read property 'getRandomBase64String' of undefined] ERROR

### SDK version number

@aws-sdk/client-location ^3.321.1

### Which JavaScript Runtime is this issue in?

React Native

### Details of the browser/Node.js/ReactNative version

node: 16.20.0

### Reproduction Steps

```
const locationClient = new LocationClient({
  credentials,
  region: 'ap-south-1',
});

const params = {
  IndexName: 'placesAPI',
  Position: [-122.33, 47.6],
};

const command = new SearchPlaceIndexForPositionCommand(params);

const data = await locationClient.send(command);
```

### Observed Behavior

When I'm trying to send a command, getting the error.

Connected these libs:

import 'react-native-get-random-values';
import 'react-native-url-polyfill/auto';
import 'url-polyfill';

Didn't help

### Expected Behavior

Returned the location object from the service.

**Comment by RanVaknin** [comment]
Hi @fruity4pie ,

Since you are using expo and react, can you please include a github repo with the necassary steps needed to run your code and observe the error youre running into? 

Thanks,
Ran~

**Comment by fruity4pie** [comment]
@RanVaknin just do `npx create-expo-app --template` install aws-location ctrl+c -> ctrl+v the code above -> check result.

**Comment by phatmann** [comment]
I ran into this bug and I solved it by switching to the JSC engine for iOS:

```
{
  "expo": {
    "jsEngine": "hermes",
    "ios": {
      "jsEngine": "jsc"
    }
  }
}
```

**Comment by fruity4pie** [comment]
@phatmann this isn't solution, because hermes works faster and produces less code. I guess jsc will be deprecated in the next expo versions.

**Comment by phatmann** [comment]
@fruity4pie For me using Hermes is not important, and I figure there are other people in the same situation as me that just want to get the builds working again. That is why I posted the workaround. This said, I want to use Hermes if I can, so I hope someone can provide a solution!

**Comment by phatmann** [comment]
Here is a [related issue](https://github.com/expo/expo/issues/17270).

**Comment by LinusU** [comment]
For reference, this is the code that `react-native-get-random-values` runs:

```js
global.ExpoModules.ExpoRandom.getRandomBase64String(10)
```

And here is an Expo maintainer confirming that it works in a clean Expo project:

https://github.com/expo/expo/issues/17270#issuecomment-1498774816

So something more is needed to trigger this, maybe the Hermes thing 🤔 

I think that the best way to solve this would be to provide a minimum viable test case which just calls `global.ExpoModules.ExpoRandom.getRandomBase64String` and submit that to the Expo team.

**Comment by fruity4pie** [comment]
It is very strange.

I did clean install and was able to run aws. But in my prev setup was expo-router. Maybe it breaks smth. Gonna check tomorrow clean setup + expo-router.

Also noticed that in expo 49 expo-random will be deprecated.

**Comment by phatmann** [comment]
For me the problem is that the segment module includes uuid.js which imports react-native-get-random-values. The segment module lists react-native-get-random-values as a peer dependency, but I had never installed it. Perhaps another module was depending on it so I did not notice. Once I added react-native-get-random-values as dependency to my project, then I get the "getRandomBase64String" error.

**Comment by berlinbra** [comment]
> For me the problem is that the segment module includes uuid.js which imports react-native-get-random-values. The segment module lists react-native-get-random-values as a peer dependency, but I had never installed it. Perhaps another module was depending on it so I did not notice. Once I added react-native-get-random-values as dependency to my project, then I get the "getRandomBase64String" error.

This actually works; removing the dependency from the project and letting the peer dependency handle it. I was experiencing this error through the react-native-gifted-chat library which uses react-native-get-random-values. Oddly, I had this dependency manually added, and didn't see any issues until recently, but removing it has resolved the issue

**Comment by phatmann** [comment]
If I remove the dependency, then uuid.js in the segment module fails to build. Glad it worked for you!

**Comment by aleqsio** [comment]
Yeah, If someone can capture the reproduction in which this call `global.ExpoModules.ExpoRandom.getRandomBase64String(10)
` fails I'd be happy to take a look. 

In general though I'd suggest to use `expo-crypto` as this provides new fast cryptographic functions using new Expo modules API and typed arrays without any conversion in JSland :) 

There's a web polyfill as well:
https://github.com/expo/expo/tree/main/packages/expo-standard-web-crypto

@LinusU Perhaps you'd be up to moving to the [expo-crypto](https://docs.expo.dev/versions/latest/sdk/crypto/) package in https://github.com/LinusU/react-native-get-random-values as well?

**Comment by RanVaknin** [comment]
Based on the discussion in this thread, this doesn't seem related to the SDK.
The AWS SDK cannot guarantee compatibility with all 3rd party tools such as Hermes.

Unless you can provide a minimal repo that can consistently raise this issue (ideally a project with only the absolute necessary imports) , I think that this question is more appropriate to be directed at the Expo / Hermes maintainers. 

Thanks,
Ran~

**Comment by phatmann** [comment]
I 100% agree @RanVaknin. I am not even using AWS SDK, but I ran into this issue and cannot find it raised anywhere else. It is specific to my Mac dev environment, as other developers from my team don't seem to be running into it! Furthermore, it happens on iOS and Android builds.

**Comment by LinusU** [comment]
> @LinusU Perhaps you'd be up to moving to the [expo-crypto](https://docs.expo.dev/versions/latest/sdk/crypto/) package in https://github.com/LinusU/react-native-get-random-values as well?

Will do, would you mind reviewing this for me? https://github.com/LinusU/react-native-get-random-values/pull/45 🙏

**Comment by phatmann** [comment]
@LinusU the fix works! Thanks so much!

**Comment by nickdep217** [comment]
The original error I got was: 
"TypeError: Cannot read property 'getRandomBase64String' of undefined, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes"

I was running a development build from expo. I was installing segment analytics.

I ran the following two commands which fixed my problems:
* npx expo install expo-random
* eas build -p ios --clear-cache

Found this at: https://docs.statsig.com/client/reactNativeExpoSDK
